lru.go

```diff
@@ -15,6 +15,7 @@ type Cache struct {
 	evictList *list.List
 	items     map[interface{}]*list.Element
 	lock      sync.RWMutex
+	OnEvicted func(key interface{}, value interface{})
```

Comment:
This is pretty not idiomatic to expose this publicly. I'd add a new constructor `NewWithEvict` that takes this function, and the existing constructor should call that with a nil argument. This way the internal methods do not need to be exposed.